README.md

```diff
@@ -160,21 +143,104 @@ try {
+
+#### What exactly is blocked by a top-level `await`?
+
+When one module imports another one, the importing module will only start executing its module body once the exporting module's body has finished executing. If the exporting module reaches a top-level await, that will have to complete before the importing module's body starts executing.
+
+#### Why doesn't top-level `await` block the import of an adjacent module?
+
+If one module wants to declare itself dependent on another module, for the purposes of waiting for that other module to complete its top-level `await` statements before the module body executes, it can delcare that other module as an import.
+
+In a case such as the following, `"Y"` will be printed on the console before `"X"`, because importing one module "before" another does not create an implicit dependency.
+
+```mjs
+// x.mjs
+await new Promise(r => setTimeout(r, 1000));
+console.log("X");
+
+// y.mjs
+console.log("Y");
+
+// z.mjs
+import "./x.mjs";
+import "./y.mjs";
+```
+
+Dependencies are required to be explicitly noted in order to boost the potential for parallelism: Most setup work that will be blocking due to a top-level await (for example, all of the case studies above) can be done in parallel with other setup work from unrelated modules. When some of this work may be highly parallelizable (e.g., network fetches), it's important to get as many of these queued up close to the start of execution as possible.
 
-#### Halting Progress
+#### What is guaranteed about code execution order?
 
-Variant A would halt progress in the module graph until resolved.
+Regardless of whether top-level `await` is used, modules always start running in the same post-order traversal established in ES2015: execution of module bodies starts with the deepest imports, in the order that the import statements for them are reached. After a top-level `await` is reached, control may be passed to start the next module in this traversal order.
 
-Variant B offers a unique approach to blocking, as it will not block siblings execution.
+#### Do these guarantees meet the needs of polyfills?
 
-The Optional Constraint would alleviate concerns of halting process.
+Currently (in a world without top-level `await`), polyfills are synchronous. So, the idiom of importing a polyfill (which modifies the global object) and then importing a module which should be affected by the polyfill will still work if top-level `await` is added. However, if a polyfill includes a top-level `await`, it will need to be imported by modules that depend on it in order to reliably take effect.
```

Comment:
I'm glad to read this; this is an important guarantee to be preserving.

Reply:
Yeah, I think this property was never in doubt, and there were just some communication issues about it.